yi-core/src/Yi/Completion.hs

```diff
@@ -56,11 +58,26 @@ infixUptoEndMatch needle haystack = case T.breakOn needle haystack of
 -- | A simple fuzzy match algorithm. Example: "abc" matches "a1b2c"
 subsequenceMatch :: String -> String -> Bool
 subsequenceMatch needle haystack = go needle haystack
-  where go (n:ns) (h:hs) | n == h = go ns hs
-        go (n:ns) (h:hs) | n /= h = go (n:ns) hs
-        go [] _ = True
-        go _ [] = False
-        go _ _  = False
+  where
+    go (n:ns) (h:hs) | n == h = go ns hs
+    go (n:ns) (h:hs) | n /= h = go (n:ns) hs
+    go [] _ = True
+    go _ [] = False
+    go _ _  = False
+
+-- | A simple fuzzy match algorithm. Example: "abc" matches "a1b2c"
+subsequenceTextMatch :: Text -> Text -> Bool
+subsequenceTextMatch needle haystack =
+  if | T.null needle   -> True
```

Comment:
The multi-way if can be avoided by just using guards:
http://learnyouahaskell.com/syntax-in-functions#guards-guards